src/rgw/rgw_op.cc

```diff
@@ -901,7 +901,8 @@ void rgw_build_iam_environment(rgw::sal::Store* store,
     s->env.emplace("aws:UserAgent", i->second);
   }
 
-  i = m.find("QUERY_STRING"); {
+  i = m.find("QUERY_STRING"); 
+  if (i != m.end()) {
```

Comment:
Add comment stating the reason for checking m.end()

Reply:
Done